### PR TITLE
tind_auto option for when last write was incomplete

### PR DESCRIPTION
**Pull request by johnomotani**
If BOUT++ gets killed while writing output, the final timestep may not be written to all files. `collect()` has an option `tind_auto` to handle this case, by truncating all the data to the shortest time from any output file. This PR adds this option to `BoutOutputs` and `squashoutput()`.

Also adds `bout-squashoutput` (copied from `BOUT-dev/bin/bout-squashoutput`) as an `entry_point` in `setup.py` so it can be used as an executable when `boutdata` is installed, independently of `BOUT-dev`. I'm not sure if the autocompletion is working properly - it doesn't work for me although from a brief google I think it's supposed to, but I never had `argcomplete` set up before so I don't know if I broke anything.

### CODE REVIEW

**Comment by johnomotani** [comment]
The `tind_auto` argument has been tested by @hahahasan through `bout-squashoutput`, which calls `squashoutput()`, which passes the argument to `BoutOutputs`. So the new code should be working correctly.

File: boutdata/squashoutput.py [hunk 1]
```diff
@@ -22,7 +22,8 @@
 def squashoutput(datadir=".", outputname="BOUT.dmp.nc", format="NETCDF4", tind=None,
                  xind=None, yind=None, zind=None, xguards=True, yguards="include_upper",
                  singleprecision=False, compress=False, least_significant_digit=None,
-                 quiet=False, complevel=None, append=False, delete=False):
+                 quiet=False, complevel=None, append=False, delete=False,
+                 tind_auto=False):
```
Comment (loeiten) [inline boutdata/squashoutput.py:26]: Would it not make more sense to default to `True` given the new error thrown in `boutdata/data.py`?

Comment (johnomotani) [reply boutdata/squashoutput.py:26]: No - `tind_auto` (maybe not the best name, but I'm copying the argument that was already there in `collect()`) is only useful if there was an error writing the output in the first place - it's a work-around. `tind` is more useful (for general slicing in the t-dimension) and can only be used when `tind_auto` is `False`.

**Comment by ZedThree** [comment]
I was going to merge in `master`, but the `main` function for `squashoutput` has now been duplicated: `squashoutput` has a version, and this PR adds `scripts/bout_squashoutput`. They're basically the same, except that the first has a bit more `description`, and the second has the `tind_auto` argument.

Any preferences on which location to keep? I'm leaning towards the `scripts` location.

**Comment by johnomotani** [comment]
I'll merge into the `scripts/bout-squashoutput` version.

**Comment by johnomotani** [comment]
In principle it should be fairly straightforward to add tests for `tind_auto` now. I don't have time to do it at the moment though... If we do want to add tests, it might be better to merge #37 and #40 first to minimise conflicts.

**Review by ZedThree — APPROVED** [review]

**Pull request merged**